test/user.test.js

```diff
@@ -1919,6 +1920,89 @@ describe('User', function() {
             expect(token).to.exist();
           });
       });
+
+      it('forwards the "options" argument to user.save() ' +
+        'when adding verification token', function() {
+        const ctxOptions = {testFlag: true};
+        const verifyOptions = {
+          type: 'email',
+          from: 'noreply@myapp.org',
+        };
+
+        // before save operation hook to capture ctx.options when saving
+        // verification token in user instance
+        User.observe('before save', function(ctx, next) {
+          if (!ctx.isNewInstance) {
+            // not checking equality since other properties are added by user.save()
+            expect(ctx.options).to.contain({testFlag: true});
```

Comment:
This is fragile. If the hook is not called then the assertion is never executed.

Here is a better way for testing the context passed to hooks: save the context to an array of observed contexts in the hook callback/handler and then assert the content of observed contexts after `user.verify` finished.

BTW you can install the observer after `User.create` finished, that way you don't have to check for `isNewInstance`.